src/Tools/MicroBuild/run-pdbgit.ps1

```diff
@@ -0,0 +1,84 @@
+<#
+
+    Run PDB Git on our PDB files to enable proper source server support
+
+#>
+Param(
+    [string]$sourcePath = $null,
+    [string]$binariesPath = $null
+)
+
+set-strictmode -version 2.0
+$ErrorActionPreference="Stop"
+
+function Get-PackagesPath {
+    $packagesPath = $env:NUGET_PACKAGES
+    if ($packagesPath -eq $null) {
+        $packagesPath = join-path $env:UserProfile ".nuget\packages\"
+    }
+
+    return $packagesPath
+}
+
+try
+{
+    [xml]$deps = get-content (join-path $sourcePath "build\Targets\Dependencies.props")
+    $pdbGitVersion = $deps.Project.PropertyGroup.PdbGitVersion
+    $packagesPath = Get-PackagesPath
+    $pdbGit = join-path $packagesPath "pdbgit\$pdbGitVersion\tools\PdbGit.exe"
+
+    # Linking our PDBs to Git is really only interesting for PE / PDB that are 
+    # being shipped to customers.  For local developement the original PDB 
```

Comment:
Yes, generated code should be embedded in the PDB. Note however that the debugger in VS 2017 doesn't support reading it from the PDB yet. Perhaps in Update 1. But we can store it there anyways. It's trivial to write a command line tool to extract the source manually when needed.